scripts/machine-launch-script.sh

```diff
@@ -1,85 +1,265 @@
-sudo yum install -y expat-devel libusb1-devel ncurses-devel cmake "perl(ExtUtils::MakeMaker)"
-# deps for poky
-sudo yum install -y python36 patch diffstat texi2html texinfo subversion chrpath git wget
-# deps for qemu
-sudo yum install -y gtk3-devel
-# deps for firesim-software (note that rsync is installed but too old)
-sudo yum install -y python36-pip python36-devel rsync
-# Install GNU make 4.x (needed to cross-compile glibc 2.28+)
-sudo yum install -y centos-release-scl
-sudo yum install -y devtoolset-8-make
-
-# install DTC
-sudo yum -y install dtc
-
-# get a proper version of git
-sudo yum -y remove git
-sudo yum -y install epel-release
-sudo yum -y install https://repo.ius.io/ius-release-el7.rpm
-sudo yum -y install git224
-
-# install verilator
-if ! command -v verilator &> /dev/null
-then
-    git clone http://git.veripool.org/git/verilator
-    cd verilator/
-    git checkout v4.034
-    autoconf && ./configure && make -j4 && sudo make install
-    cd ..
-fi
-
-# bash completion for manager
-sudo yum -y install bash-completion
-
-# graphviz for manager
-sudo yum -y install graphviz python-devel
-
-# used for CI
-sudo yum -y install expect
-
-# upgrade pip
-sudo pip3 install --upgrade pip==21.3.1
-# install requirements
-sudo python3 -m pip install fab-classic==1.19.1
-sudo python3 -m pip install boto3==1.20.21
-sudo python3 -m pip install colorama==0.4.3
-sudo python3 -m pip install argcomplete==1.12.3
-sudo python3 -m pip install graphviz==0.19
-# for some of our workload plotting scripts
-sudo python3 -m pip install pyparsing==3.0.6
-sudo python3 -m pip install numpy==1.19.5
-sudo python3 -m pip install kiwisolver==1.3.1
-sudo python3 -m pip install matplotlib==3.3.4
-sudo python3 -m pip install pandas==1.1.5
-sudo python3 -m pip install awscli==1.22.21
-sudo python3 -m pip install pytest==6.2.5
-sudo python3 -m pip install pytest-dependency==0.5.1
-sudo python3 -m pip install moto==2.2.17
-sudo python3 -m pip install sure==2.0.0
-# needed for the awstools cmdline parsing
-sudo python3 -m pip install pyyaml==5.4.1
-
-# setup argcomplete
-activate-global-python-argcomplete
-
-} 2>&1 | tee /home/centos/machine-launchstatus.log
-
-# get a regular prompt
-echo "PS1='\u@\H:\w\\$ '" >> /home/centos/.bashrc
-echo "machine launch script completed" >> /home/centos/machine-launchstatus
+
+    if [[ ! -r /etc/os-release ]]; then
+	echo "::ERROR:: $0 depends on /etc/os-release for distro-specific setup and it doesn't exist here"
+	exit 1
+    fi
+
+    if [[ "$(uname -io)" != "x86_64 GNU/Linux" ]]; then
+	echo "::ERROR:: $0 only supports 'x86_64 GNU/Linux' not '$(uname -io)'"
+	exit 1
+    fi
+
+    OS_FLAVOR=$(grep '^ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"')
+    OS_VERSION=$(grep '^VERSION_ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"')
+
+    echo "machine launch script started" > machine-launchstatus
+    chmod ugo+r machine-launchstatus
+
+    # platform-specific setup 
```

Comment:
So you said this was here as a demonstration right? You're going to remove it before PR?

Reply:
We could, or we could leave it in there so that there is a way for people to suggest that they need a new package and we can incrementally work on getting it added into conda so that it is distro agnostic?

If we remove it, then we should rework https://github.com/firesim/firesim/pull/986/files#diff-457f33389bd30641349e0da5cbe159fde6ae8455c42a051c0f723136c08577ebR146-R148

Reply:
Then i think we should maybe make the warning an error right?

Reply:
Sure.  It's probably better to force someone to at least add the new platform and have the section empty if it doesn't need to have something.